dspace-api/src/main/java/org/dspace/statistics/SolrLoggerServiceImpl.java

```diff
@@ -985,28 +996,29 @@ protected String getDateView(String name, String type, Context context) {
     @Override
     public QueryResponse query(String query, String filterQuery,
                                String facetField, int rows, int max, String dateType, String dateStart,
-                               String dateEnd, List<String> facetQueries, String sort, boolean ascending)
+                               String dateEnd, List<String> facetQueries, String sort, boolean ascending,
+                               int facetMinCount)
         throws SolrServerException, IOException {
         if (solr == null) {
             return null;
         }
 
         // System.out.println("QUERY");
         SolrQuery solrQuery = new SolrQuery().setRows(rows).setQuery(query)
-                                             .setFacetMinCount(1);
+                                             .setFacetMinCount(facetMinCount);
```

Comment:
facetMinCount needed to be able to set to 0:
* To be able to get the months which have no views as point => need to set facetMinCount to 0 for just that query (here:https://github.com/atmire/DSpace/blob/w2p-71214_Usage-reports/dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/UsageReportService.java#L223), which required the change in SolrLoggerServiceImpl to make facetMinCount a method parameter, but this did propagate into adding this method parameter into a lot of other place that use SolrLoggerServiceImpl.  So now points with views=0 also get returned (for TotalVisits I could manually create a point with view = 0, but this was needed just for TotalVisitsPerMonth, not for the Cities/Countries, otherwise there would be long list of points with views=0 that have a view in another item, so for those facetMinCount=1 was still needed
* Note: if you want for the TotalDownloads on an item have a point for each of its bitstream (no just those with views > 0), then the point creation just has to be changed to check totalVisits on each bitstream because if you use facetMinCount=0 there then every bitstream ever retrieved will be in points list